main.tf

```diff
@@ -5,11 +5,11 @@ locals {
 }
```

Comment:
The `validate_demo` check uses `tobool` against an error message.   I realize this wasn't part of the PR, but I'll note what this looked like when I ran from examples/demo_cluster with `TF_VAR_deploy_demo=true`.

```
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Invalid function argument
│ 
│   on ../../main.tf line 4, in locals:
│    4:   validate_demo = (var.deploy_demo == true && var.global_ip == false) ? tobool("Demo is only deployed if global_ip = true.") : true
│     ├────────────────
│     │ while calling tobool(v)
│ 
│ Invalid value for "v" parameter: cannot convert "Demo is only deployed if global_ip = true." to bool; only the strings "true" or "false" are allowed.
╵
```

To get around this, I set `TF_VAR_global_ip=true`. 

Possibly unrelated, I ran into this error and needed to run with `-auto-approve`:
```
Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: ╷
│ Error: error asking for approval: EOF
```

Reply:
The EOF problem was with my terminal window. unrelated.

Reply:
Switching to terraform 1.9 that allows cross object referencing for input variable validations https://www.hashicorp.com/blog/terraform-1-9-enhances-input-variable-validations